### ISSUE TITLE
Issue with the ranges

### ISSUE THREAD

**Issue by cesalo** [JibbyJames/data-studio-lighthouse-gauge#11]
First of all thanks for putting this together. Great work.
The issue I'm finding is the following (maybe is just a misunderstanding of how to use the ranges):
These are my settings
![image](https://user-images.githubusercontent.com/16859916/113509556-50afec80-9524-11eb-8e4b-a939607e5b0d.png)
If I have a value in the limit of the mid section is not going into that color:
![image](https://user-images.githubusercontent.com/16859916/113509596-750bc900-9524-11eb-96ee-2faa488e8207.png)
In order to work I need to put the mid section higher and overlapping with the high section:
![image](https://user-images.githubusercontent.com/16859916/113509631-ab494880-9524-11eb-9d13-9a5739c5271a.png)
and it works:
![image](https://user-images.githubusercontent.com/16859916/113509637-b4d2b080-9524-11eb-9346-2c6bd4e32156.png)
so my questions is if that how it was created where I need to overlap the end of the prev range to the start of the next one?
Thanks in advance.

**Comment by JibbyJames** [comment]
Hi Cesalo,

Thanks for the feedback!

I actually encountered this myself quite recently, and I wasn't sure if it
always worked that way or not, as I have made a few changes around the
range logic.

What I assumed was happening is the metric value being processed is
actually something like 0.885 which is above 0.88 but below 0.89, so sits
in between the defined ranges, so an overlap is required. This makes sense
to me, but I haven't checked if that's actually what the metric value is.
The lower range is evaluated first so that's another factor that goes into
what value to set as the overlap.

Thanks,
James
